src/common/turbo.ts

```diff
@@ -47,31 +58,57 @@ export async function signedRequestHeadersFromSigner({
   signer: TurboArNSSigner;
   nonce?: string;
 }): Promise<RawAxiosRequestHeaders> {
-  await (signer as ArconnectSigner).setPublicKey?.();
-  const signature = await signer.sign(Uint8Array.from(Buffer.from(nonce)));
-
-  let publicKey: string;
-  switch (signer.signatureType) {
-    case SignatureConfig.ARWEAVE:
-      publicKey = toB64Url(signer.publicKey);
-      break;
-    case SignatureConfig.ETHEREUM:
-      publicKey = '0x' + signer.publicKey.toString('hex');
-      break;
-    // TODO: solana sig support
-    // case SignatureConfig.SOLANA:
-    // case SignatureConfig.ED25519:
-    default:
-      throw new Error(
-        `Unsupported signer type for signing requests: ${signer.signatureType}`,
-      );
-  }
+  let signature: string | undefined = undefined;
+  let publicKey: string | undefined = undefined;
+
```

Comment:
```suggestion
if (signer instanceof ArconnectSigner) signer = ArconnectSigner.signer;
```

Reply:
Not a fan of overriding arguments, but this accomplishes the same thing: [35ae6aa](https://github.com/ar-io/ar-io-sdk/pull/458/commits/35ae6aa2f2e7a5429c35df9a4a4cabbbb499fed3)